components/clarity-repl/src/repl/clarity_values.rs

```diff
@@ -22,47 +22,52 @@ pub fn uint8_to_value(mut value: &[u8]) -> Value {
 
 pub fn value_to_string(value: &Value) -> String {
     match value {
-        Value::Principal(principal_data) => {
-            format!("'{principal_data}")
-        }
+        Value::Principal(principal_data) => format!("'{}", principal_data),
         Value::Tuple(tup_data) => {
-            let mut data = Vec::new();
-            for (name, value) in tup_data.data_map.iter() {
-                data.push(format!("{}: {}", &**name, value_to_string(value)))
-            }
-            format!("{{ {} }}", data.join(", "))
+            let data = tup_data
+                .data_map
+                .iter()
+                .map(|(name, value)| format!("{}: {}", name, value_to_string(value)))
+                .collect::<Vec<_>>()
```

Comment:
this is doing a lot of allocations for no good reason, first in each format, then in the collect, and finally again in join.

You could instead setup an empty mut string and write! to it, which would only allocate the string

Reply:
Addressed